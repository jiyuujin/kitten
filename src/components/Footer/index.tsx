import React, { FC } from 'react'
import { useIntl } from 'gatsby-plugin-react-intl'
import * as SC from './index.module.scss'

const Footer: FC = () => {
  const intl = useIntl()

  return (
    <footer className={SC.footer}>
      <p className={SC.contact}>
        <a
          href="https://docs.google.com/forms/d/e/1FAIpQLSfFzwsCVnFbu-lV0Jz2fEYHR97hpBSK2g6kEwc-G1fo976ngA/viewform"
          target="_blank"
          rel="noopener noreferrer"
        >
          {intl.formatMessage({ id: 'contact' })}
        </a>
      </p>
      <p className={SC.copyright}>Created Â© 2020 jiyuujin LAB. All Rights Reserved.</p>
    </footer>
  )
}

export default Footer
