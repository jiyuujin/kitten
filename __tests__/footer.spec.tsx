import React from 'react'
import renderer from 'react-test-renderer'
import Footer from '../src/components/Footer'

jest.mock('react-intl', () => {
    const reactIntl = jest.requireActual('react-intl')
    const intl = reactIntl.createIntl({
        locale: 'en',
    })
    return {
        ...reactIntl,
        useIntl: () => intl,
    }
})

describe('Footer', () => {
    it('render correctly', () => {
        const tree = renderer.create(<Footer />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
