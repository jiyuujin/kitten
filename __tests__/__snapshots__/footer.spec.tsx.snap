// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer render correctly 1`] = `
<footer>
  <p>
    <a
      href="https://webneko.dev/contact"
      rel="noopener noreferrer"
      target="_blank"
    >
      CONTACT
    </a>
  </p>
  <p>
    Created Â© 2020 jiyuujin LAB. All Rights Reserved.
  </p>
</footer>
`;
